src/components/Layout/DefaultLayout/DefaultAxis.js

```diff
@@ -2,50 +2,115 @@ import { useDroppable } from '@dnd-kit/core'
+    let hasDuplicates = false
+    const dataElements = new Map()
 
-    // TODO - using the rawDimensionId instead of dimensionId
-    // is a temporary workaround
-    // until the backend is updated to return programStageId.dimensionId
-    // in analytics response.metadata.items
-    const getDimension = (id) => {
-        let dimension
-        if (metadata[id]?.name) {
+    const dimensions = dimensionIds.map((id) => {
+        let dimension = {}
+        if (metadata[id]) {
             dimension = metadata[id]
         } else {
             const [rawDimensionId] = id.split('.').reverse()
             dimension = metadata[rawDimensionId]
         }
+        return dimension
+    })
+
+    if (inputType === OUTPUT_TYPE_ENROLLMENT) {
+        dimensions.forEach((dimension) => {
+            const dataElementId = dimension.id.split('.')[1]
+            if (
+                dimension.dimensionType === DIMENSION_TYPE_DATA_ELEMENT &&
+                dataElementId
+            ) {
+                const dataElementCount = dataElements.get(dataElementId)
+
+                if (dataElementCount) {
+                    dataElements.set(dataElementId, dataElementCount + 1)
+                    hasDuplicates = true
+                } else {
+                    dataElements.set(dataElementId, 1)
+                }
+            }
+        })
 
-        return dimension || {}
+        return hasDuplicates
+            ? dimensions.map((dimension) => {
+                  const [programStageId, dataElementId] =
+                      dimension.id.split('.')
+                  if (dataElementId && dataElements.get(dataElementId) > 1) {
+                      dimension.stageName =
+                          programStageNames?.get(programStageId)
+                  }
+                  return dimension
+              })
+            : dimensions
     }
 
-    const activeIndex = draggingId ? axis.indexOf(draggingId) : -1
+    return dimensions
+}
+
+const DefaultAxis = ({ axisId, className }) => {
+    const lastDropZoneId = getDropzoneId(axisId, LAST)
+    const { over, setNodeRef } = useDroppable({
+        id: lastDropZoneId,
+    })
+
+    const dispatch = useDispatch()
+    const dimensionIds = useSelector((state) =>
+        sGetUiDimensionIdsForLayoutAxis(state, axisId)
+    )
+    const renderChips = useSelector(renderChipsSelector)
+    const draggingId = useSelector(sGetUiDraggingId)
+    const metadata = useSelector(sGetMetadata)
+    const inputType = useSelector(sGetUiInputType)
+    const selectedProgramId = useSelector(sGetUiProgramId)
+    const program = useSelector((state) =>
+        sGetMetadataById(state, selectedProgramId)
+    )
+
+    const programStageNames = useMemo(
+        () =>
+            program?.programStages?.reduce((acc, stage) => {
+                acc.set(stage.id, stage.name)
+                return acc
+            }, new Map()),
+        [program]
+    )
+
+    const dimensions = getDimensionsWithStageName({
```

Comment:
same as above, any particular reason for keeping this fn outside of the `DefaultAxis` scope or can it just be put inside the scope so e.g. `metadata` can be used directly instead of first being defined above and then be passed in?